File: Java/Java Kasus/src/com/myclass/Barang.java
package com.myclass;

abstract class Barang {
    private String namaBarang;
    private int hargaBarang, stokBarang;

    public abstract String displayBarang();

    Barang() {
    }

    protected Barang(String namaBarang, int hargaBarang, int stokBarang) {
        this.namaBarang = namaBarang;
        this.hargaBarang = hargaBarang;
        this.stokBarang = stokBarang;
    }

    protected void tambahStok() {
        this.stokBarang += 1;
        System.out.println("Stok barang berhasil ditambahkan menjadi : " + this.stokBarang + "\n");
    }

    protected void tambahStok(int jmlTambah) {
        this.stokBarang += jmlTambah;
        System.out.println("Stok barang berhasil ditambahkan menjadi : " + this.stokBarang + "\n");
    }

    protected void kurangStok() {
        this.stokBarang -= 1;
        System.out.println("Stok barang berhasil dikurangkankan menjadi : " + this.stokBarang + "\n");
    }

    protected void kurangStok(int jmlKurang) {
        this.stokBarang -= jmlKurang;
        System.out.println("Stok barang berhasil dikurangkankan menjadi : " + this.stokBarang + "\n");
    }

    protected boolean stokBarang() {
        if (this.stokBarang > 0)
            return true;
        else
            return false;
    }

    protected boolean stokBarang(boolean status) {
        if (status) {
            if (this.stokBarang > 0)
                return true;
            else
                return false;
        } else {
            if (this.stokBarang < 0)
                return false;
            else
                return true;
        }
    }

    protected void infoBarang() {
        System.out.print("Nama Barang : " + this.namaBarang + ", Harga Barang : Rp." + this.hargaBarang
                + ", Stok Barang : " + this.stokBarang + " pcs, ");
    }

    public void setNamaBarang(String namaBarang) {
        this.namaBarang = namaBarang;
    }

    public void setHargaBarang(int hargaBarang) {
        this.hargaBarang = hargaBarang;
    }

    public void setStokBarang(int stokBarang) {
        this.stokBarang = stokBarang;
    }

    public String getNamaBarang() {
        return this.namaBarang;
    }

    public int getHargaBarang() {
        return this.hargaBarang;
    }

    public int getStokBarang() {
        return this.stokBarang;
    }
}

File: C++ Kasus/src/Barang.cpp
#include "myclass/Barang.h"
#include <iostream>

Barang::Barang(std::string inputNama, int inputHarga, int inputStok)
{
	this->namaBarang = inputNama;
	this->hargaBarang = inputHarga;
	this->stokBarang = inputStok;
}

void Barang::tambahBarang()
{
	stokBarang += 1;
	std::cout << "Stok berhasil ditambahkan menjadi : " << stokBarang << std::endl;
}

void Barang::tambahBarang(int jmlTambah)
{
	stokBarang += jmlTambah;
	std::cout << "Stok berhasil ditambahkan menjadi : " << stokBarang << std::endl;
}

void Barang::kurangBarang()
{
	stokBarang -= 1;
	std::cout << "Stok berhasil dikurangkan menjadi : " << stokBarang << std::endl;
}

void Barang::kurangBarang(int jmlKurang)
{
	stokBarang -= jmlKurang;
	std::cout << "Stok berhasil dikurangkan menjadi : " << stokBarang << std::endl;
}

bool Barang::cekBarang()
{
	if (stokBarang > 0)
		return true;
	else
		return false;
}

bool Barang::cekBarang(bool status)
{
	if (status)
	{
		if (stokBarang > 0)
			return true;
		else
			return false;
	}
	else
	{
		if (stokBarang > 0)
			return false;
		else
			return true;
	}
}

void Barang::infoBarang()
{
	std::cout << "Nama Barang : " << this->namaBarang << ", Harga Barang : Rp." << this->hargaBarang << ", Stok Barang : " << this->stokBarang;
}

void Barang::setNamaBarang(std::string namaBarang)
{
	this->namaBarang = namaBarang;
}

void Barang::setHargaBarang(int hargaBarang)
{
	this->hargaBarang = hargaBarang;
}

void Barang::setStokBarang(int stokBarang)
{
	this->stokBarang = stokBarang;
}

std::string Barang::getNamaBarang()
{
	return this->namaBarang;
}

int Barang::getHargaBarang()
{
	return this->hargaBarang;
}

int Barang::getStokBarang()
{
	return this->stokBarang;
}